core/src/components/footer/footer.tsx

```diff
@@ -24,9 +35,46 @@ export class Footer implements ComponentInterface {
    */
   @Prop() translucent = false;
 
+  componentDidLoad() {
+    this.checkCollapsibleFooter();
+  }
+
+  componentDidUpdate() {
+    this.checkCollapsibleFooter();
+  }
```

Comment:
Since this can run multiple times on the same component, does the scroll listener added in `checkCollapsibleFooter` need to be removed first to dedupe? Or is that handled through some other mechanism 🤔 (Also applies to `ion-header`)

Reply:
Oops yes we should destroy the scroll listener so there is no duplication. The header already has this logic: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/header/header.tsx#L68-L69